e2e/evm_methods_test.go

```diff
@@ -0,0 +1,247 @@
+//go:build e2e
+
+package e2e
+
+import (
+	"time"
+
+	"github.com/buildwithgrove/path/qos/jsonrpc"
+)
+
+/* -------------------- EVM JSON-RPC Method Definitions -------------------- */
+
+// Docs reference for all methods:
+// https://ethereum.org/en/developers/docs/apis/json-rpc/
+const (
+	eth_blockNumber           jsonrpc.Method = "eth_blockNumber"
+	eth_call                  jsonrpc.Method = "eth_call"
+	eth_getTransactionReceipt jsonrpc.Method = "eth_getTransactionReceipt"
+	eth_getBlockByNumber      jsonrpc.Method = "eth_getBlockByNumber"
+	eth_getBalance            jsonrpc.Method = "eth_getBalance"
+	eth_chainId               jsonrpc.Method = "eth_chainId"
+	eth_getTransactionCount   jsonrpc.Method = "eth_getTransactionCount"
+	eth_getTransactionByHash  jsonrpc.Method = "eth_getTransactionByHash"
+	eth_gasPrice              jsonrpc.Method = "eth_gasPrice"
+)
+
+// runAllMethods returns all EVM JSON-RPC methods for a service load test
+func runAllMethods() []jsonrpc.Method {
+	return []jsonrpc.Method{
+		eth_blockNumber,
+		eth_call,
+		eth_getTransactionReceipt,
+		eth_getBlockByNumber,
+		eth_getBalance,
+		eth_chainId,
+		eth_getTransactionCount,
+		eth_getTransactionByHash,
+		eth_gasPrice,
+	}
+}
+
+type (
+	// methodDefinition contains all configuration and test requirements for a method
+	methodDefinition struct {
+		methodConfig
+		methodSuccessRates
+	}
+
+	// methodConfig contains the configuration for a method to be tested.
+	// This includes the total number of requests to send & the requests per second.
+	methodConfig struct {
+		totalRequests int // Total number of requests to send
+		rps           int // Requests per second
+	}
+
+	// methodSuccessRates contains the minimum success rate and maximum
+	// latency requirements for a method to pass the load test.
+	methodSuccessRates struct {
+		successRate   float64       // Minimum success rate (0-1)
+		maxP50Latency time.Duration // Maximum P50 latency
+		maxP95Latency time.Duration // Maximum P95 latency
+		maxP99Latency time.Duration // Maximum P99 latency
+	}
+)
+
+// methodDefinitions contains all method definitions for a service load test.
+// this allows customizing the configuration for each method as desired.
+var methodDefinitions = map[jsonrpc.Method]methodDefinition{
```

Comment:
Not a blocker, but this looks like a good candidate for reading the settings from a yaml file.

Reply:
Sure that could be a good future feature but not sure that it adds a lot of value currently, what about adding it as a TODO_FUTURE or TODO_IMPROVE?